torch/onnx/_internal/exporter.py

```diff
@@ -0,0 +1,230 @@
+    ) -> None:
+        """Saves the in-memory ONNX model to ``destination`` using specified ``serializer``.
+        If no ``serializer`` is specified, the model will be serialized as Protobuf."""
+
+        if serializer is None:
+            serializer = ProtobufExportOutputSerializer()
+        if isinstance(destination, str):
+            with open(destination, "wb") as f:
+                serializer.serialize(self, f)
+        else:
+            serializer.serialize(self, destination)
+
+
+class Exporter(abc.ABC):
+    @_beartype.beartype
+    def __init__(
+        self,
+        options: ExportOptions,
+        model: torch.nn.Module,
+        model_args: Sequence[Any],
+        model_kwargs: Mapping[str, Any],
+    ):
+        self.options = options
+        self.model = model
+        self.model_args = model_args
+        self.model_kwargs = model_kwargs
+
+    @abc.abstractmethod
+    def run(self) -> ExportOutput:
+        pass
+
+    @property
+    def logger(self) -> logging.Logger:
+        # options.logger will always be resolved to an instance when constructing
+        assert isinstance(self.options.logger, logging.Logger)
+        return self.options.logger
+
+
+@_beartype.beartype
+def _resolve_export_options(options: Optional[ExportOptions]) -> ExportOptions:
+    if options is None:
+        options = ExportOptions()
+
+    T = TypeVar("T")
+
+    def resolve(value: Optional[T], fallback: Callable[[], T]):
+        return fallback() if value is None else value
+
+    return ExportOptions(
+        opset_version=resolve(options.opset_version, lambda: _DEFAULT_OPSET_VERSION),
+        dynamic_shapes=options.dynamic_shapes,
+        logger=resolve(
+            options.logger, lambda: logging.getLogger().getChild("torch.onnx")
+        ),
+    )
+
+
+@_beartype.beartype
+def dynamo_export(
+    model: torch.nn.Module,
+    /,
+    *model_args,
+    export_options: Optional[ExportOptions] = None,
+    **model_kwargs,
+) -> ExportOutput:
+    """Export a torch.nn.Module to an ONNX graph.
+
+    Args:
+        model: The PyTorch model to be exported to ONNX.
+        model_args: Positional inputs to ``model``.
+        model_kwargs: Keyword inputs to ``model``.
+        export_options: Options to influence the export to ONNX.
+
+    Returns:
+        An in-memory representation of the exported ONNX model.
+    """
+
+    resolved_export_options = _resolve_export_options(export_options)
+
+    def assert_dependencies():
+        # this has been resolved above, so will not be 'None'
+        assert (opset_version := resolved_export_options.opset_version) is not None
+
+        # TODO: each of these imports/assertions should be turned into
+        # an actionable diagnostic, e.g.:
+        # - Please `pip install onnxscript_torchlib`
+        # - You have specified an opset version X that is newer than the
+        #   supported opset version Y
+        # - etc.
+        import onnx
```

Comment:
```suggestion
        try:
             import onnx
         except ModuleNotFoundError:
             resolved_export_options.logger.error("ONNX not found. Run `pip install onnx` before!")             
```

Reply:
I do not want to do this right now. The entire `assert_dependencies` function will be rewritten as part of @BowenBao's diagnostics API pass.